internal/shader/syntax_test.go

```diff
@@ -1320,12 +1320,30 @@ func TestSyntaxOperatorShift(t *testing.T) {
 		stmt string
 		err  bool
 	}{
-		{stmt: "a := 1 << 2.0; _ = a", err: true},
-		{stmt: "a := 1.0 << 2; _ = a", err: true},
-		{stmt: "a := 1.0 << 2.0; _ = a", err: true},
+		// {stmt: "s := 1; var a float = float(1 << s); _ = a", err: true},
+		// {stmt: "s := 1; var a float = float(1.0 << s); _ = a", err: true},
+		// {stmt: "s := 1; var a int = int(1 << s); _ = a", err: false},
+		// {stmt: "s := 1; var a int = int(1.0 << s); _ = a", err: false},
+		// {stmt: "s := 1; a := 1 << s; _ = a", err: false},
+		// {stmt: "s := 1; a := 1.0 << s; _ = a", err: true},
+		// {stmt: "s := 1; a := int(1.0 << s); _ = a", err: false},
+		// {stmt: "s := 1; var a float = float(1.0 << s); _ = a", err: true},
+		// {stmt: "s := 1; var a float = 1.0 << s; _ = a", err: true},
+		// {stmt: "s := 1; var a float = 1 << s; _ = a", err: true},
+		// {stmt: "s := 1; var a int = 1.0 << s; _ = a", err: false},
+		// {stmt: "s := 1; var a int = 1 << s; _ = a", err: false},
+
+		{stmt: "var a float = 1.0 << 2.0; _ = a", err: false},
+		{stmt: "var a int = 1.0 << 2; _ = a", err: false},
+		{stmt: "var a float = 1.0 << 2; _ = a", err: false},
+		{stmt: "var a = 1.0 << 2; _ = a", err: false},
```

Comment:
Isn't this the same as `a := 1.0 << 2`?

Reply:
Yes, if used inside a function body.
https://go.dev/play/p/I5OIJe7WjOD

Reply:
So this is redundant. We already have `{stmt: "a := 1.0 << 2; _ = a", err: false}`.